src/targets/services/categorize.js

```diff
@@ -36,19 +33,26 @@ export const categorize = async () => {
     }
   } else {
     // Apply filters first
-    let filteredOperations = operations
-    if (filters.date) {
-      filteredOperations = filteredOperations.filter(
-        e =>
-          new Date(filters.date).valueOf() -
-            new Date(e.cozyMetadata.createdAt).valueOf() <=
-          0
-      )
+    let filtersToApply = {}
+
+    if (filters.minOperationDate) {
+      filtersToApply = Object.assign(filtersToApply, {
+        date: { $gt: filters.minOperationDate }
+      })
     }
 
-    model = Model.fromDocs(filteredOperations)
+    model = Model.fromDocs(
+      await client.queryAll(
+        Q(BANK_DOCTYPE)
+          .where(filtersToApply)
+          .sortBy([{ date: 'asc' }])
+      )
+    )
   }
 
+  // Fetch data
+  const { data: operations } = await client.query(Q(BANK_DOCTYPE))
```

Comment:
If you need all the operations, shouldn't you just run `await client.queryAll(Q(BANK_DOCTYPE).sortBy([{ date: 'asc' }]))` and then filter the operations you need for the model (with date >  minOperationDate ) ? 
Like discussed earlier?